File: Native/Test/Fabric/remove-scope.js
F = require('Fabric').createClient();

node = F.DG.createNode('node');
eventHandler = F.DG.createEventHandler('eventHandler');
eventHandler.setScope('node', node);
console.log((function (scopes) {
  var result = [];
  for (var scope in scopes)
    result.push(scope);
  return result;
})(eventHandler.getScopes()));
try {
  eventHandler.removeScope('notANode');
  F.flush();
}
catch (e) {
  console.log("eventHandler.removeScope('notANode'): " + e);
}
eventHandler.removeScope('node');
console.log(eventHandler.getScopes());

F.close();

File: Native/Test/Python/remove-scope.py
#
#  Copyright 2010-2012 Fabric Engine Inc. All rights reserved.
#

import fabric
F = fabric.createClient()

node = F.DG.createNode('node')
eventHandler = F.DG.createEventHandler('eventHandler')
eventHandler.setScope('node', node)

def printScopes( scopes ):
  result = []
  for scope in scopes:
    result.append( scope )
  print result

printScopes( eventHandler.getScopes() )

try:
  eventHandler.removeScope('notANode')
  F.flush()
except Exception as e:
  print("eventHandler.removeScope('notANode'): " + str(e))

eventHandler.removeScope('node')
print(eventHandler.getScopes())

F.close()
